include/boost/wintls/context.hpp

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright (c) 2020 Kasper Laudrup (laudrup at stacktrace dot dk)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_WINTLS_CONTEXT_HPP
+#define BOOST_WINTLS_CONTEXT_HPP
+
+#include <boost/wintls/method.hpp>
+
+#include <boost/wintls/detail/config.hpp>
+#include <boost/wintls/detail/context_certificates.hpp>
+
+#include <boost/winapi/handles.hpp>
+
+#include <memory>
+#include <string>
+
+extern "C" {
+  struct CERT_CONTEXT__;
+  typedef CERT_CONTEXT CERT_CONTEXT_;
+}
+
+namespace boost {
+namespace wintls {
+
+namespace detail {
+class sspi_handshake;
+}
+
+class context {
+public:
+  /** Construct a context.
+   *
+   * @param connection_method The @ref method to use for connections.
+   */
+  explicit context(method connection_method)
+    : ctx_certs_(std::make_unique<detail::context_certificates>())
+    , method_(connection_method)
+    , verify_server_certificate_(false) {
```

Comment:
Isn't it a bit dangerous to have this default off?
Most other systems I know have this default on.